src/main/java/duke/SaveHandler.java

```diff
@@ -0,0 +1,94 @@
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try {
+            File file = new java.io.File( "./duke.txt");
+            Scanner fileReader = new Scanner(file);
+            while (fileReader.hasNextLine()) {
+                String data = fileReader.nextLine();
+                tasks.add(stringToTask(data));
+            }
+        } catch (FileNotFoundException fe) {
+            File file = new java.io.File( "./duke.txt");
+            try {
+                file.createNewFile();
+            } catch (IOException ioe) {
+                System.out.println("Error creating new data file.\n"+ioe);
+                System.exit(-1);
+            }
+        } catch (StreamCorruptedException e) {
+            System.out.println("Corrupted data file.");
+            System.exit(-1);
+        }
+
+        return tasks;
+    }
+
+    public void save(ArrayList<Task> list) {
+        StringBuilder data = new StringBuilder();
+        for (Task task : list) {
+            data.append(taskToString(task));
+        }
+
+        try {
+            FileWriter writer = new FileWriter("./duke.txt");
+            writer.write(data.toString());
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Error saving data file.");
+            System.exit(-1);
+        }
+    }
+
+    private Task stringToTask(String taskString) throws StreamCorruptedException {
+        String[] parts = taskString.split("/");
+        if (Objects.equals(parts[0], "T")) {
+            Todo task = new Todo(parts[2]);
+            if (Objects.equals(parts[1], "1")) {
+                task.setIsDone(true);
+            }
+            return task;
+        } else if (Objects.equals(parts[0], "D")) {
+            Deadline task = new Deadline(parts[2], parts[3]);
+            if (Objects.equals(parts[1], "1")) {
+                task.setIsDone(true);
+            }
+            return task;
+        } else if (Objects.equals(parts[0], "E")) {
+            Event task = new Event(parts[2], parts[3], parts[4]);
+            if (Objects.equals(parts[1], "1")) {
+                task.setIsDone(true);
+            }
+            return task;
+        } else {
+            throw new StreamCorruptedException();
+        }
+    }
+
+    private String taskToString(Task task) {
+        String output = "";
+        switch (task.getClass().getName()) {
+            case "duke.Task":
+            case "duke.Todo":
+                output = "T/" + (task.isDone ? "1/" : "0/") + task.description + "\n";
+                break;
+            case "duke.Deadline":
+                output = "D/" + (task.isDone ? "1/" : "0/") + task.description + "/"
+                        + ((Deadline) task).by + "\n";
+                break;
+            case "duke.Event":
+                output = "E/" + (task.isDone ? "1/" : "0/") + task.description + "/"
+                        + ((Event) task).by + "/" + ((Event) task).from + "\n";
+                break;
```

Comment:
Instead of using a switch case to generate an output of each of the task, you might want to consider using the polymorphism aspect of Java to write a more scalable product. To simply put, each type of `task` should know how to print themselves within their class instead of checking their task type and to provide a different output in this `taskToString()` method.